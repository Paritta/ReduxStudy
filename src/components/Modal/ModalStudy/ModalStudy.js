import React from "react";
import styled, {keyframes} from "styled-components";
import { fadeIn } from 'react-animations';
import PropTypes from "prop-types";
import ModalStudyCommentContainer from "../../../containers/Modalcontainers/ModalStudyCommentContainer";
import ModalBoxLeftPage from "../ModalBoxLeftPage/ModalBoxLeftPage";
import EmptyImage from "../../../asset/EmptyImage.png";
import oc from "open-color";
import ReactLoading from "react-loading";

const propTypes = {
    Modal: PropTypes.shape({
        modalProps: PropTypes.shape({
            postId: PropTypes.string.isRequired,
            data: PropTypes.shape({
                username: PropTypes.string
            })
        })
    }),
    auth: PropTypes.object,
    showModal: PropTypes.func,
    postDeleteRequest: PropTypes.func,
    imageReceiveRequest: PropTypes.func,
    ImageReceive: PropTypes.object,
    heartSendRequest: PropTypes.func,
    heartDeleteRequest: PropTypes.func,
};

const defaultTypes = {
    Modal: {},
    auth: {},
    showModal() {},
    postDeleteRequest() {},
    imageReceiveRequest() {},
    ImageReceive: {},
    heartSendRequest() {},
    heartDeleteRequest() {},
};

const fadeInAnimation = keyframes`${fadeIn}`;

const Wrapper = styled.div`
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    
    overflow: hidden;
    
    z-index: 10;
    
    width: 70%;
`;

const AnimationWrapper = styled.div`
    display: flex;
    justify-content: space-between;
    
    animation: 1s ${fadeInAnimation};
`;

const ModalBoxLeft = styled.div`
    position: relative;

    background: white;
    border: 1px solid rgba(0, 0, 0, 0.3);
    
    width: 58%;
    height: 800px;
`;

const ModalBoxLeftHeader = styled.div`
    position: absolute;
    top: 0;
    left: 0;
    
    background: url(${props => props.ImageUrl ? props.ImageUrl : EmptyImage}) center center;
    background-size: cover;
    
    width: 100%;
    height: 60%;
`;

const ModalBoxRight = styled.div`
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.3);
    
    width: 35%;
`;

const Dimmed = styled.div`
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    
    z-index: 5;
    
    background: rgba(0, 0, 0, 0.3);
`;

const ImageRegister = styled.button`
    position: absolute;
    
    top: 3%;
    right: 3%;
    
    font-weight: 400;
    font-size: 1.2em;
    
    padding: 0.25em 1em;
    
    background: palevioletred;
    color: white;
    
    border: 2px solid palevioletred;
    
    transition: all 1s ease;
    
    &:hover {
        background: ${oc.pink[4]};
        border: 2px solid ${oc.pink[4]};
    }
`;

const WrappedReactLoading = styled.div`
    transform: scale(2);
    position: absolute;
    
    top: 15%;
    left: 40%;
    
    padding: 80px 0;
`;

export class ModalStudy extends React.Component {
    componentDidMount () {
        this.props.imageReceiveRequest(this.props.Modal.modalProps.postId)
    }

    render () {
        const { Modal, showModal, auth, postDeleteRequest, ImageReceive, heartSendRequest, heartDeleteRequest }  = this.props;
        const modalProps = Modal.modalProps;
        const author = modalProps.data.author;
        const Pending = ImageReceive.pending;
        let ImageUrl = false;
        const HeartCount = modalProps.data.HeartCount;

        if(ImageReceive.data !== undefined && ImageReceive.data.length > 0) {
            ImageUrl = ImageReceive.data[ImageReceive.data.length-1].Url;
        }

        return (
            <div>
                <Dimmed />
                <Wrapper>
                    <AnimationWrapper>
                        <ModalBoxLeft>
                            {/* 이미지 로딩 */}
                            {
                                Pending &&
                                <WrappedReactLoading>
                                    <ReactLoading type="cylon" color="palevioletred"/>
                                </WrappedReactLoading>
                            }

                            {/* 이미지가 없을 때는 EmptyImage  */}
                            {
                                auth !== null && modalProps.data.PostImageKey === "" &&
                                <ModalBoxLeftHeader
                                    ImageUrl={false}
                                >
                                    {
                                        auth.uid === author &&
                                        <ImageRegister
                                            onClick={() => showModal({
                                                modalType: "MODAL_IMAGE",
                                                modalProps: modalProps.postId
                                            })}
                                        >
                                            사진 등록
                                        </ImageRegister>
                                    }
                                </ModalBoxLeftHeader>
                            }

                            {/* 이미지가 있을 때는 ImageUrl, 이미지 등록 버튼 삭제 */}
                            {
                                modalProps.data.PostImageKey !== "" &&
                                <ModalBoxLeftHeader
                                    ImageUrl={ImageUrl}
                                >
                                </ModalBoxLeftHeader>
                            }
                            <ModalBoxLeftPage
                                PageData={modalProps}
                                auth={auth}
                                postDeleteRequest={postDeleteRequest}
                                heartSendRequest={heartSendRequest}
                                heartDeleteRequest={heartDeleteRequest}
                                postId={modalProps.postId}
                            />
                        </ModalBoxLeft>
                        <ModalBoxRight>
                            <ModalStudyCommentContainer
                                postId={modalProps.postId}
                                username={modalProps.data.username}
                                auth={auth}
                                author={author}
                                HeartCount={HeartCount}
                            />
                        </ModalBoxRight>
                    </AnimationWrapper>
                </Wrapper>
            </div>
        )
    }
}

ModalStudy.propTypes = propTypes;
ModalStudy.defaultTypes = defaultTypes;

export default ModalStudy
