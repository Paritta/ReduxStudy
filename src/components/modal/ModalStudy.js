import React from "react";
import styled, {keyframes} from "styled-components";
import { fadeIn } from 'react-animations';
import PropTypes from "prop-types";
import ModalStudyForm from "./ModalStudyForm";

const propTypes = {
    Modal: PropTypes.object,
    hideModal: PropTypes.func,
};

const defaultTypes = {
    hideModal() {},
};

const fadeInAnimation = keyframes`${fadeIn}`;

const Wrapper = styled.div`
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    border-radius: 10px;
    
    overflow: hidden;
    
    z-index: 10;
    
    width: 550px;
`;

const AnimationWrapper = styled.div`
    animation: 1s ${fadeInAnimation};
`;

const ModalHeader = styled.div`
    font-style: normal;
    font-weight: 400;
    color: white;
    padding: 15px 15px;
    background: palevioletred;
`;

const ModalBox = styled.div`
    background: white;
    border: 1px solid rgba(0, 0, 0, 0.3);
    padding: 15px;
`;

const Dimmed = styled.div`
    position: fixed;
    top: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    
    z-index: 5;
    
    background: rgba(0, 0, 0, 0.3);
`;

class ModalStudy extends React.Component {
    render () {
        const { hideModal } = this.props;

        return (
            <div>
                a
            </div>
        )
    }
};

ModalStudy.propTypes = propTypes;
ModalStudy.defaultTypes = defaultTypes;

export default ModalStudy
