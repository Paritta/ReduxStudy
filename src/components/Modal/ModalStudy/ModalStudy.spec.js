import React from "react";
import ReactTestRenderer from "react-test-renderer"
import { ModalStudy } from "./ModalStudy";
import "jest-styled-components";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {};
const store = mockStore(initialState);

const Modal = {
    modalProps: {
        postId: "",
        data: {
            username: "",
            values: {
                Category: "",
                introduction: "웹 해킹 스터디",
                Location: "중앙도서관 하늘 정원",
                Number: "8명",
                StudyTitle: "웹 해킹",
                Time : "매주 월요일 7시",
            }
        },
        author: ""
    }
};

it("ModalStudy render correctly", () => {
    const tree = ReactTestRenderer.create(
        <Provider store={store}>
            <ModalStudy
                Modal={ Modal }
            />
        </Provider>
    );
    expect(tree.toJSON()).toMatchSnapshot();
});
