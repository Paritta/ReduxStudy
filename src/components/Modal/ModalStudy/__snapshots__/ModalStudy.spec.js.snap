// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalStudy render correctly 1`] = `
.c0 {
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  z-index: 5;
  background: rgba(0,0,0,0.3);
}

.c1 {
  position: fixed;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  overflow: hidden;
  z-index: 10;
  width: 70%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-animation: 1s eMLfYp;
  animation: 1s eMLfYp;
}

.c3 {
  position: relative;
  background: white;
  border: 1px solid rgba(0,0,0,0.3);
  width: 58%;
  height: 800px;
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  background: url(ModalStudyAsset.jpeg) left center;
  background-size: cover;
  width: 100%;
  height: 60%;
}

.c5 {
  position: absolute;
  top: 60%;
  height: 40%;
  width: 100%;
}

.c6 {
  margin: 30px 0 15px 0;
}

.c7 {
  font-weight: 500;
  font-size: 0.9em;
  border-radius: 5px;
  height: 20px;
  width: 20px;
  background: #e9ecef;
  color: black;
  padding: 5px 10px 5px 10px;
  margin: 9px 9px 9px 0;
}

.c8 {
  font-weight: 700;
  font-size: 2.5em;
}

.c9 {
  margin: 50px;
  font-weight: 300;
  font-size: 1.5em;
}

.c10 {
  font-weight: 500;
  font-size: 0.9em;
  border-radius: 5px;
  display: inline-block;
  background: #e9ecef;
  color: black;
  padding: 5px 10px 5px 10px;
  margin: 9px 9px 9px 0;
}

.c11 {
  padding-left: 5px;
}

.c12 {
  font-weight: 500;
  font-size: 0.9em;
  border-radius: 5px;
  display: inline-block;
  background: #e9ecef;
  color: black;
  padding: 5px 10px 5px 10px;
  margin: 9px 9px 9px 0;
}

.c13 {
  background: white;
  border: 1px solid rgba(0,0,0,0.3);
  width: 35%;
}

<div>
  <div
    className="c0"
  />
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        />
        <div
          className="c5"
        >
          <div
            className="c6"
          >
            <span
              className="c7"
            >
              
            </span>
            <span
              className="c7"
            >
              8명
            </span>
          </div>
          <div
            className="c8"
          >
            웹 해킹
          </div>
          <div
            className="c9"
          />
          <div
            className="c10"
          >
            <svg
              fill="currentColor"
              height={25}
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "color": "Black",
                  "verticalAlign": "middle",
                }
              }
              viewBox="0 0 40 40"
              width={25}
            >
              <g>
                <path
                  d="m20 19.1q1.7 0 2.9-1.2t1.2-2.9-1.2-2.9-2.9-1.2-2.9 1.2-1.2 2.9 1.2 2.9 2.9 1.2z m0-15.7q4.8 0 8.2 3.4t3.4 8.2q0 2.4-1.2 5.5t-2.9 5.9-3.4 5.1-2.8 3.8l-1.3 1.3q-0.5-0.5-1.2-1.4t-2.9-3.6-3.5-5.2-2.8-5.8-1.2-5.6q0-4.8 3.4-8.2t8.2-3.4z"
                />
              </g>
            </svg>
            <span
              className="c11"
            >
              중앙도서관 하늘 정원
            </span>
          </div>
          <div
            className="c12"
          >
            <svg
              fill="currentColor"
              height={25}
              preserveAspectRatio="xMidYMid meet"
              style={
                Object {
                  "color": "Black",
                  "verticalAlign": "middle",
                }
              }
              viewBox="0 0 40 40"
              width={25}
            >
              <g>
                <path
                  d="m20.9 11.6v8.8l7.5 4.4-1.3 2.2-8.7-5.4v-10h2.5z m-0.9 21.8q5.5 0 9.4-4t4-9.4-4-9.4-9.4-4-9.4 4-4 9.4 4 9.4 9.4 4z m0-30q6.9 0 11.8 4.8t4.8 11.8-4.8 11.8-11.8 4.8-11.8-4.8-4.8-11.8 4.8-11.8 11.8-4.8z"
                />
              </g>
            </svg>
            <span
              className="c11"
            >
              매주 월요일 7시
            </span>
          </div>
        </div>
      </div>
      <div
        className="c13"
      />
    </div>
  </div>
</div>
`;
